ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx

```diff
@@ -56,44 +83,120 @@ const WalletPaymentPickPspScreen = () => {
     }, [dispatch, walletId, paymentAmountInCents])
   );
 
+  React.useEffect(
+    () => () => {
+      dispatch(walletPaymentResetPickedPsp());
+    },
+    [dispatch]
+  );
+
+  const handleChangePspSorting = (sortType: WalletPaymentPspSortType) => {
+    setShowFeaturedPsp(sortType === "default");
+    dispatch(walletPaymentSortPsp(sortType));
```

Comment:
The sort should be made on the UI, and not on data layer.
We should manage this with a `sortedPspList` directly in the screen component.

Reply:
I agree! Addressed into 487939547b2f2447a9e68139c288641a345a67ef 👍